compiler/rustc_privacy/src/lib.rs

```diff
@@ -232,11 +221,21 @@ where
                     return ControlFlow::Continue(());
                 }
 
-                self.def_id_visitor.visit_def_id(
-                    data.def_id,
-                    "associated type",
-                    &LazyDefPathStr { def_id: data.def_id, tcx },
-                )?;
+                let kind = match kind {
+                    ty::Inherent => Some("associated type"),
+                    ty::Weak => Some("type alias"),
+                    // Trait associated types don't have visibility, they are
+                    // visible if their parent trait is.
+                    ty::Projection => None,
```

Comment:
```suggestion
                    ty::Projection => Some("associated type"),
```
Not sure what the last commit intends to demonstrate.
I didn't suggest skipping this logic for associated types, I only suggested to use their own def ids.

Associated type def ids do have (semantic) visibility (the comment above is wrong in this regard), but that visibility is indeed inherited from the trait (the comment above is right in that regard).

Reply:
yea that makes sense. I added a change that just exposes associated types directly here. I will edit git history at the end before merging the PR